ldurniat/berry.lua

```diff
@@ -191,7 +195,7 @@ end
 -- @return The newly created image sheet.
 -- 
 -- Original code from https://github.com/ponywolf/ponytiled 
------------------------------------------------------------------------------------------------   
+--------------------------------------------------------------------------------   
 local function loadTileset( tileset, dir, tileId )
```

Comment:
So turns out the `loadTileset` function was doing multiple things depending on the arguments a user would provide.  It would either:

- create a image sheet and return it
- return the directory, width, and height on an image based on tile_id

Functions and methods should do one thing.  (a lot of people call it the [KISS pinciple](https://en.wikipedia.org/wiki/KISS_principle)) The main reason behind this is because the code is a lot easier to reuse when it's in a small state and not coupled or tangled together.  Since I am trying to add TexturePacker functionality to this bit of code, I've had to untangle it and clean it up.  To do that I have taken the function and split it into smaller functions that have a simple purpose.  Now we have,

`getImageSheet(tileset)`
`getTileImage(tileset, tile_id)`

On top of that you can run your logic loop to determine which one you need to use,

```lua
if tileset_has_embedded_images then
  getTileImage(...)
elseif tileset_has_grouped_image then
  getImageSheet(...)
elseif tileset_has_texturepacker_images then
  getTP(...)
```